### PR TITLE
Jhbate/plus oauth

### PR DESCRIPTION
**Pull request by jh-bate**
@jebeck includes the addition of oauth2 functionality. Lets talk this through and review

### CODE REVIEW

File: oauth2/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+```
+client_id=d995aaabe7
+client_secret=dd85baba6e0816662e4b16d32242f6526cf3e35a
+```
+
+### Notes:
+
+* What is a platfrom user?
+ * This is your applications account on the Tidepool platfrom
```
Comment (jebeck) [inline oauth2/docs/README.md]: Spelling: should be `application's`

File: oauth2/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+Make a note of both your client_id and client_secret.
+
+```
+client_id=d995aaabe7
+client_secret=dd85baba6e0816662e4b16d32242f6526cf3e35a
+```
+
+### Notes:
+
+* What is a platfrom user?
```
Comment (jebeck) [inline oauth2/docs/README.md]: `platform`

File: oauth2/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ * This is your applications account on the Tidepool platfrom
+
+* What is the redirect URI?
+ * The redirect URI is the URL within your application that will receive the OAuth2 credentials.
+
+* Scopes available:
+  * Requests uploading of data on behalf
```
Comment (jebeck) [inline oauth2/docs/README.md]: Clearer way to say this? Feels like it just cuts off... Requests uploading on behalf of who?

**Comment by jebeck** [comment]
Some spelling etc. comments on the OAuth docs README. Otherwise, I looked through with an eye to verifying that no functionality changed in the pre-existing endpoint. I believe that's true - all the changes look merely organizational and/or clarifying, no changes to functionality. So LGTM.

**Comment by jh-bate** [comment]
Have addressed some of the OAuth docs issues - there is more to come before making this all more  public so will be a WIP

**Pull request merged**